### PR TITLE
Add context for the modal categories

### PR DESCRIPTION
**Pull request by alejandrocabriales**
# Ticket

https://trello.com/c/8LcgvKYi/286-user-story-basic-list-with-ecosystem-actors

# What solved
- [X] Add Context for show modal in all the views, to make it more accessible and easy to use

# Description
This adds context to the modal of categories

### CODE REVIEW

File: src/stories/containers/FinancesOverview/FinancesOverviewContainer.tsx [hunk 1]
```diff
@@ -54,84 +55,82 @@ const FinancesOverviewContainer: React.FC<FinancesOverviewContainerProps> = ({
...
+        </Container>
+        <CategoryModalComponent
+          checkOut={checkOut}
+          headCountCategories={headCountCategories}
+          notHeadCountCategory={noHeadCountCategories}
+          handleCloseModal={handleCloseModal}
+          handleCheckedExpandedAll={handleCheckedExpandedAll}
+          handleChangeItemAccordion={handleChangeItemAccordion}
+          isLight={isLight}
+          openModal={openModal}
```
Comment (YasielCabrera) [inline src/stories/containers/FinancesOverview/FinancesOverviewContainer.tsx:131]: Except, maybe the `openModal` prop, all the other props could be accessed through the hook inside the modal

File: src/stories/containers/TransparencyReport/components/ExpenseReport/ExpenseReport.tsx [hunk 2]
```diff
@@ -29,16 +32,11 @@ interface ExpenseReportProps {
...
-const ExpenseReport: React.FC<ExpenseReportProps> = ({
-  currentMonth,
-  budgetStatements,
-  code,
-  longCode,
-  expenseCategories,
-}) => {
+const ExpenseReport: React.FC<ExpenseReportProps> = ({ currentMonth, budgetStatements, code, longCode }) => {
+  const { checkOut, handleChangeItemAccordion, handleCheckedExpandedAll, headCountCategories, noHeadCountCategories } =
+    useCategoriesContextModal();
```
Comment (YasielCabrera) [inline src/stories/containers/TransparencyReport/components/ExpenseReport/ExpenseReport.tsx:39]: Note that once you move the invocation of this hook to inside the modal, this is no needed at all here

File: src/stories/containers/TransparencyReport/components/TransparencyActuals/TransparencyActuals.tsx [hunk 3]
```diff
@@ -28,6 +28,16 @@ interface Props {
...
+  const {
+    checkOut,
+    handleChangeItemAccordion,
+    handleCheckedExpandedAll,
+    handleCloseModal,
+
+    headCountCategories,
+    noHeadCountCategories,
+    openModal,
+  } = useCategoriesContextModal();
```
Comment (YasielCabrera) [inline src/stories/containers/TransparencyReport/components/TransparencyActuals/TransparencyActuals.tsx:40]: As before, this should no be needed once you move the hook to inside the modal

File: src/stories/containers/TransparencyReport/components/TransparencyActuals/useTransparencyActuals.ts [hunk 4]
```diff
@@ -16,14 +16,23 @@ import {
...
+  const {
+    checkOut,
+    handleChangeItemAccordion,
+    handleCheckedExpandedAll,
+    handleCloseModal,
+    handleOpenModal,
+    headCountCategories,
+    noHeadCountCategories,
+    openModal,
+  } = useCategoriesContextModal();
```
Comment (YasielCabrera) [inline src/stories/containers/TransparencyReport/components/TransparencyActuals/useTransparencyActuals.ts:35]: same

**Review by YasielCabrera — CHANGES_REQUESTED** [review]

File: src/core/context/CategoryModalContext.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  handleCheckedExpandedAll: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleChangeItemAccordion: (id: string, expanded: boolean) => void;
+};
+
+const CategoriesModalContext = createContext<CategoryModalContextValues>({} as CategoryModalContextValues);
+const useCategoriesContextModal = () => useContext(CategoriesModalContext);
+
+interface Props extends React.PropsWithChildren {
+  expenseCategories: ExpenseCategory[];
+  children: ReactNode;
```
[the remark this answers is not in the document]
Comment (alejandrocabriales) [reply src/core/context/CategoryModalContext.tsx]: you are right , done

File: src/core/context/CategoryModalContext.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+  checkOut: boolean;
+  headCountCategories: ParsedExpenseCategoryWithExpanded[];
+  noHeadCountCategories: ParsedExpenseCategoryWithExpanded[];
+  handleCheckedExpandedAll: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  handleChangeItemAccordion: (id: string, expanded: boolean) => void;
+};
+
+const CategoriesModalContext = createContext<CategoryModalContextValues>({} as CategoryModalContextValues);
+const useCategoriesContextModal = () => useContext(CategoriesModalContext);
```
[the remark this answers is not in the document]
Comment (alejandrocabriales) [reply src/core/context/CategoryModalContext.tsx]: you are right, done

File: src/stories/containers/FinancesOverview/FinancesOverviewContainer.tsx [hunk 7]
```diff
@@ -54,84 +55,82 @@ const FinancesOverviewContainer: React.FC<FinancesOverviewContainerProps> = ({
...
-    byCategoryBreakdownExpenses,
-    expenseCategories
-  );
+    handleCheckedExpandedAll,
+    handleCloseModal,
+    handleOpenModal,
+    headCountCategories,
+    noHeadCountCategories,
+    openModal,
+  } = useCategoriesContextModal();
```
[the remark this answers is not in the document]
Comment (alejandrocabriales) [reply src/stories/containers/FinancesOverview/FinancesOverviewContainer.tsx]: you are right, done

File: src/stories/containers/FinancesOverview/FinancesOverviewContainer.tsx (hunk 1 above)
Comment (alejandrocabriales) [reply src/stories/containers/FinancesOverview/FinancesOverviewContainer.tsx]: you are right, all are except the handleOpenModal

File: src/stories/containers/TransparencyReport/components/TransparencyActuals/TransparencyActuals.tsx (hunk 3 above)
Comment (alejandrocabriales) [reply src/stories/containers/TransparencyReport/components/TransparencyActuals/TransparencyActuals.tsx]: done

File: src/stories/containers/TransparencyReport/components/TransparencyActuals/useTransparencyActuals.ts (hunk 4 above)
Comment (alejandrocabriales) [reply src/stories/containers/TransparencyReport/components/TransparencyActuals/useTransparencyActuals.ts]: done

File: src/stories/containers/TransparencyReport/components/ExpenseReport/ExpenseReport.tsx (hunk 2 above)
Comment (alejandrocabriales) [reply src/stories/containers/TransparencyReport/components/ExpenseReport/ExpenseReport.tsx]: done

File: pages/index.tsx [hunk 8]
```diff
@@ -35,13 +36,15 @@ const FinanceOverviewPage: NextPage<FinanceOverviewPageProps> = ({
...
-      byCategoryBreakdownExpenses={byCategoryBreakdownExpenses}
-      expenseCategories={expenseCategories}
-    />
+    <ModalCategoriesProvider expenseCategories={expenseCategories}>
+      <FinancesOverviewContainer
+        quarterExpenses={quarterExpenses}
+        monthlyExpenses={monthlyExpenses || []}
+        byBudgetBreakdownExpenses={byBudgetBreakdownExpenses}
+        byCategoryBreakdownExpenses={byCategoryBreakdownExpenses}
+        expenseCategories={expenseCategories}
```
Comment (YasielCabrera) [inline pages/index.tsx:45]: if you passed the `expenseCategories` in the provider, do you need to pass it as props to this component?

File: src/stories/containers/FinancesOverview/useFinancesOverview.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import { useMediaQuery } from '@mui/material';
-import { useModalCategory } from '@ses/components/BasicModal/useModalCategory';
+// import { useModalCategory } from '@ses/components/BasicModal/useModalCategory';
```
Comment (YasielCabrera) [inline src/stories/containers/FinancesOverview/useFinancesOverview.ts:2]: is this comment needed?

File: src/stories/containers/FinancesOverview/useFinancesOverview.ts [hunk 10]
```diff
@@ -12,7 +12,7 @@ import {
 } from './utils/costBreakdown';
 import { parseQuarter } from './utils/quarters';
 import type { CostBreakdownFilterValue, ExtendedExpense } from './financesOverviewTypes';
-import type { ExpenseCategory } from '@ses/core/models/dto/expenseCategoriesDTO';
+// import type { ExpenseCategory } from '@ses/core/models/dto/expenseCategoriesDTO';
```
Comment (YasielCabrera) [inline src/stories/containers/FinancesOverview/useFinancesOverview.ts:15]: is this comment needed?

**Review by YasielCabrera — CHANGES_REQUESTED** [review]

File: src/stories/containers/FinancesOverview/useFinancesOverview.ts (hunk 9 above)
Comment (alejandrocabriales) [reply src/stories/containers/FinancesOverview/useFinancesOverview.ts:2]: not, done

File: src/stories/containers/FinancesOverview/useFinancesOverview.ts (hunk 10 above)
Comment (alejandrocabriales) [reply src/stories/containers/FinancesOverview/useFinancesOverview.ts:15]: not , done

File: src/stories/containers/TransparencyReport/components/TransparencyActuals/useTransparencyActuals.ts [hunk 11]
```diff
@@ -16,14 +16,23 @@ import {
...
+  const {
+    checkOut,
+    handleChangeItemAccordion,
+    handleCheckedExpandedAll,
+    handleCloseModal,
+    handleOpenModal,
+    headCountCategories,
+    noHeadCountCategories,
+    openModal,
+  } = useCategoriesModalContext();
```
[the remark this answers is not in the document]
Comment (alejandrocabriales) [reply src/stories/containers/TransparencyReport/components/TransparencyActuals/useTransparencyActuals.ts:35]: yes, the only what I need is handleOpenModal, to open the modal

File: pages/index.tsx (hunk 8 above)
Comment (alejandrocabriales) [reply pages/index.tsx:45]: You are right, It's not need it. Done

**Review by YasielCabrera — APPROVED** [review]
LGTM

**Pull request merged**